### PR TITLE
[SPARK-40710][DOCS] Supplement undocumented parquet configurations in documentation

### PR DESCRIPTION
**Pull request by dcoliversun**
### What changes were proposed in this pull request?

This PR aims to supplement undocumented parquet configurations in documentation.

### Why are the changes needed?

Help users to confirm configurations through documentation instead of code.

### Does this PR introduce _any_ user-facing change?

Yes, more configurations in documentation.

### How was this patch tested?

Pass the GA.

### CODE REVIEW

File: docs/sql-data-sources-parquet.md [hunk 1]
```diff
@@ -454,6 +454,28 @@ Configuration of Parquet can be done using the `setConf` method on `SparkSession
   </td>
   <td>1.3.0</td>
 </tr>
+<tr>
+  <td><code>spark.sql.parquet.int96TimestampConversion</code></td>
```
Comment (dcoliversun) [inline docs/sql-data-sources-parquet.md:458]: https://github.com/apache/spark/blob/309638eeefbfb13dae8dbded0279bf44390389ee/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L899-L905

File: docs/sql-data-sources-parquet.md [hunk 2]
```diff
@@ -454,6 +454,28 @@ Configuration of Parquet can be done using the `setConf` method on `SparkSession
...
+  <td>false</td>
+  <td>
+    This controls whether timestamp adjustments should be applied to INT96 data when 
+    converting to timestamps, for data written by Impala.  This is necessary because Impala 
+    stores INT96 data with a different timezone offset than Hive & Spark.
+  </td>
+  <td>2.3.0</td>
+</tr>
+<tr>
+  <td><code>spark.sql.parquet.outputTimestampType</code></td>
```
Comment (dcoliversun) [inline docs/sql-data-sources-parquet.md:468]: https://github.com/apache/spark/blob/309638eeefbfb13dae8dbded0279bf44390389ee/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L911-L921

File: docs/sql-data-sources-parquet.md [hunk 3]
```diff
@@ -473,6 +495,17 @@ Configuration of Parquet can be done using the `setConf` method on `SparkSession
   <td>Enables Parquet filter push-down optimization when set to true.</td>
   <td>1.2.0</td>
 </tr>
+<tr>
+  <td><code>spark.sql.parquet.aggregatePushdown</code></td>
```
Comment (dcoliversun) [inline docs/sql-data-sources-parquet.md:499]: https://github.com/apache/spark/blob/309638eeefbfb13dae8dbded0279bf44390389ee/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L1003-L1010

File: docs/sql-data-sources-parquet.md [hunk 4]
```diff
@@ -493,6 +526,17 @@ Configuration of Parquet can be done using the `setConf` method on `SparkSession
   </td>
   <td>1.5.0</td>
 </tr>
+<tr>
+  <td><code>spark.sql.parquet.respectSummaryFiles</code></td>
```
Comment (dcoliversun) [inline docs/sql-data-sources-parquet.md:530]: https://github.com/apache/spark/blob/309638eeefbfb13dae8dbded0279bf44390389ee/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L872-L879

File: docs/sql-data-sources-parquet.md [hunk 5]
```diff
@@ -505,6 +549,84 @@ Configuration of Parquet can be done using the `setConf` method on `SparkSession
   </td>
   <td>1.6.0</td>
 </tr>
+<tr>
+  <td><code>spark.sql.parquet.enableVectorizedReader</code></td>
```
Comment (dcoliversun) [inline docs/sql-data-sources-parquet.md:553]: https://github.com/apache/spark/blob/309638eeefbfb13dae8dbded0279bf44390389ee/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L1033-L1038

File: docs/sql-data-sources-parquet.md [hunk 6]
```diff
@@ -505,6 +549,84 @@ Configuration of Parquet can be done using the `setConf` method on `SparkSession
...
+  <td>true</td>
+  <td>
+    Enables vectorized parquet decoding.
+  </td>
+  <td>2.0.0</td>
+</tr>
+<tr>
+  <td><code>spark.sql.parquet.enableNestedColumnVectorizedReader</code></td>
```
Comment (dcoliversun) [inline docs/sql-data-sources-parquet.md:561]: https://github.com/apache/spark/blob/309638eeefbfb13dae8dbded0279bf44390389ee/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L1040-L1046

File: docs/sql-data-sources-parquet.md [hunk 7]
```diff
@@ -505,6 +549,84 @@ Configuration of Parquet can be done using the `setConf` method on `SparkSession
...
+  <td>true</td>
+  <td>
+    Enables vectorized Parquet decoding for nested columns (e.g., struct, list, map). 
+    Requires <code>spark.sql.parquet.enableVectorizedReader</code> to be enabled.
+  </td>
+  <td>3.3.0</td>
+</tr>
+<tr>
+  <td><code>spark.sql.parquet.recordLevelFilter.enabled</code></td>
```
Comment (dcoliversun) [inline docs/sql-data-sources-parquet.md:570]: https://github.com/apache/spark/blob/309638eeefbfb13dae8dbded0279bf44390389ee/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L1048-L1056

File: docs/sql-data-sources-parquet.md [hunk 8]
```diff
@@ -505,6 +549,84 @@ Configuration of Parquet can be done using the `setConf` method on `SparkSession
...
+  <td>
+    If true, enables Parquet's native record-level filtering using the pushed down filters. 
+    This configuration only has an effect when <code>spark.sql.parquet.filterPushdown</code> 
+    is enabled and the vectorized reader is not used. You can ensure the vectorized reader 
+    is not used by setting <code>spark.sql.parquet.enableVectorizedReader</code> to false.
+  </td>
+  <td>2.3.0</td>
+</tr>
+<tr>
+  <td><code>spark.sql.parquet.columnarReaderBatchSize</code></td>
```
Comment (dcoliversun) [inline docs/sql-data-sources-parquet.md:581]: https://github.com/apache/spark/blob/309638eeefbfb13dae8dbded0279bf44390389ee/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L1058-L1063

File: docs/sql-data-sources-parquet.md [hunk 9]
```diff
@@ -505,6 +549,84 @@ Configuration of Parquet can be done using the `setConf` method on `SparkSession
...
+  <td>4096</td>
+  <td>
+    The number of rows to include in a parquet vectorized reader batch. The number should 
+    be carefully chosen to minimize overhead and avoid OOMs in reading data.
+  </td>
+  <td>2.4.0</td>
+</tr>
+<tr>
+  <td><code>spark.sql.parquet.fieldId.write.enabled</code></td>
```
Comment (dcoliversun) [inline docs/sql-data-sources-parquet.md:590]: https://github.com/apache/spark/blob/309638eeefbfb13dae8dbded0279bf44390389ee/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L1065-L1072

File: docs/sql-data-sources-parquet.md [hunk 10]
```diff
@@ -505,6 +549,84 @@ Configuration of Parquet can be done using the `setConf` method on `SparkSession
...
+  <td>true</td>
+  <td>
+    Field ID is a native field of the Parquet schema spec. When enabled, 
+    Parquet writers will populate the field Id metadata (if present) in the Spark schema to the Parquet schema.
+  </td>
+  <td>3.3.0</td>
+</tr>
+<tr>
+  <td><code>spark.sql.parquet.fieldId.read.enabled</code></td>
```
Comment (dcoliversun) [inline docs/sql-data-sources-parquet.md:599]: https://github.com/apache/spark/blob/309638eeefbfb13dae8dbded0279bf44390389ee/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L1074-L1081

File: docs/sql-data-sources-parquet.md [hunk 11]
```diff
@@ -505,6 +549,84 @@ Configuration of Parquet can be done using the `setConf` method on `SparkSession
...
+  <td>false</td>
+  <td>
+    Field ID is a native field of the Parquet schema spec. When enabled, Parquet readers 
+    will use field IDs (if present) in the requested Spark schema to look up Parquet 
+    fields instead of using column names.
+  </td>
+  <td>3.3.0</td>
+</tr>
+<tr>
+  <td><code>spark.sql.parquet.fieldId.read.ignoreMissing</code></td>
```
Comment (dcoliversun) [inline docs/sql-data-sources-parquet.md:609]: https://github.com/apache/spark/blob/309638eeefbfb13dae8dbded0279bf44390389ee/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L1083-L1090

File: docs/sql-data-sources-parquet.md [hunk 12]
```diff
@@ -505,6 +549,84 @@ Configuration of Parquet can be done using the `setConf` method on `SparkSession
...
+  <td>false</td>
+  <td>
+    When the Parquet file doesn't have any field IDs but the 
+    Spark read schema is using field IDs to read, we will silently return nulls 
+    when this flag is enabled, or error otherwise.
+  </td>
+  <td>3.3.0</td>
+</tr>
+<tr>
+  <td><code>spark.sql.parquet.timestampNTZ.enabled</code></td>
```
Comment (dcoliversun) [inline docs/sql-data-sources-parquet.md:619]: https://github.com/apache/spark/blob/309638eeefbfb13dae8dbded0279bf44390389ee/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L1092-L1101

**Review by dcoliversun — COMMENTED** [review]
cc @HyukjinKwon @dongjoon-hyun 
It would be good if you have a time to review :)

**Review by srowen — COMMENTED** [review]
I think these are OK. I don't think any are meant to be hidden or internal-only.
Are these logically ordered?

**Comment by dcoliversun** [comment]
@srowen Yes, related configurations are grouped together and are in logical order with each other.

**Pull request closed without merging**